mlflow/utils/search_utils.py

```diff
@@ -37,19 +36,41 @@ def filter_string(self):
     def search_expressions(self):
         return self._search_expressions
 
+    @classmethod
+    def _trim_ends(cls, string_value):
+        return string_value[1:-1]
+
+    @classmethod
+    def _is_quoted(cls, value, pattern):
```

Comment:
What happens if `value` is a single character?

Reply:
It should be fine. Since we call `_is_quoted` before trimming that checks for presence of both starting and ending quote and trimming an empty string should be ok. However, I will add a `len(value) >= 2` check